assets/scripts/dialogs/GeotagDialog.jsx

```diff
@@ -220,48 +199,62 @@ class GeotagDialog extends React.Component {
       true
     )
 
-    this.props.addLocation(location)
-    this.props.saveStreetName(location.hierarchy.street, false)
+    batch(() => {
+      dispatch(addLocation(location))
+      dispatch(saveStreetName(location.hierarchy.street, false))
+    })
   }
 
-  handleClearLocation = (event) => {
+  const handleClearLocation = (event) => {
     trackEvent(
       'Interaction',
       'Geotag dialog: cleared existing location',
       null,
       null,
       true
     )
-    this.props.clearLocation()
+    dispatch(clearLocation())
   }
 
-  reverseGeocode = (latlng) => {
+  const reverseGeocode = (latlng) => {
     const url = `${REVERSE_GEOCODE_ENDPOINT}&point.lat=${latlng.lat}&point.lon=${latlng.lng}`
 
     return window.fetch(url).then((response) => response.json())
   }
 
-  setSearchResults = (point, label, bbox) => {
+  // Search dialog state dosen't sync up with map state (IMO should clear or update with label)
+  const setSearchResults = (point, locationProperties) => {
     const latlng = {
       lat: point[0],
       lng: point[1]
     }
+    updateMap(latlng, locationProperties)
+  }
 
-    this.setState({
-      zoom: MAP_LOCATION_ZOOM,
-      mapCenter: latlng,
-      renderPopup: true,
-      markerLocation: latlng,
-      label: label,
-      bbox: bbox || null
-    })
+  function updateMap (latlng, locationProperties) {
+    /*
+    after the location position is updated,
+    we need to update the map UI elements
+    an update place information in the store
+    */
+    setRenderPopup(true)
+    setMarkerLocation(latlng)
+    setLabel(locationProperties.label)
+    setMapCenter(latlng)
+
+    dispatch(
+      setMapState({
+        markerLocation: latlng,
+        addressInformation: locationProperties
+      })
+    )
   }
 
   /**
    * Determines if the street location can be saved or edited.
    */
-  canEditLocation = () => {
-    const { street } = this.props
+  const canEditLocation = () => {
```

Comment:
`const` is necessary because of the transition from a class component to a function component. The previous syntax made those functions methods on the class. The current syntax requires that they be defined as a variable.

It can also be a function. I prefer to use the `function` keyword because it then becomes super obvious what it is. However, using `const` to get a one-line arrow function is also a good trade off.

Reply:
Ok, I'll give this some thought and make a call